queries/cpp/highlights.scm

```diff
@@ -49,58 +49,93 @@
 (destructor_name
   (identifier) @method)
 
+; functions
 (function_declarator
-      declarator: (qualified_identifier
-        name: (identifier) @function))
+  (qualified_identifier
+    (identifier) @function))
 (function_declarator
-      declarator: (qualified_identifier
-        name: (qualified_identifier
-          name: (identifier) @function)))
+  (qualified_identifier
+    (qualified_identifier
+      (identifier) @function)))
 (function_declarator
-      declarator: (template_function
-        name: (identifier) @function))
+  (qualified_identifier
+    (qualified_identifier
+      (qualified_identifier
+        (identifier) @function))))
+((qualified_identifier
+  (qualified_identifier
+    (qualified_identifier
+      (qualified_identifier
+        (identifier) @function)))) @_parent
+  (#has-ancestor? @_parent function_declarator))
```

Comment:
Hmm, doesn't this also run into the issue of needing a `has-ancestor?` to actually highlight properly?
There really is no beautiful way to avoid it then :')

Reply:
It'll only use `has-ancestor` if there's a node that matches 4 nested scopes for qualified identifiers.

Reply:
Let's really hope that it is rare. If there are occurrences of `has-ancestors?` the predicate could be triggered pretty often.

Reply:
If this distinction between `@function` and `@function.call` gets too complex we might want to revert to what it was before the distinction.